bot/cogs/weather_cog.py

```diff
@@ -0,0 +1,206 @@
+        # Exceptions:
+        # & _ - and ,
+        # per the ASCII Table https://www.asciitable.com
+        loc = re.sub("[^a-zA-Z0-9 ,&_-]+", "", loc)
+
+        # Geocoding URL
+        url_GeoAPI = f'https://geocode.xyz/{loc}'
+
+        self.geocode_api_key = BotSecrets.get_instance().geocode_key
+        self.weather_api_key = BotSecrets.get_instance().weather_key
+
+        geo_queryparams = {
+            'auth' : self.geocode_api_key,
+            'json' : '1',
+            }
+
+        # Try Except for catching errors that could give away either API key
+        try:
+            async with aiohttp.request("GET", url_GeoAPI, params=geo_queryparams) as response:
+                if (response.status == 200):
+                    res_geo_json = await response.json()
+                    city = res_geo_json.get('standard',{}).get('city',{})
+                    lon = res_geo_json.get('longt',{})
+                    lat = res_geo_json.get('latt',{})
+                    
+                    url = "https://api.openweathermap.org/data/2.5/onecall"
+
+                    queryparams = {
+                        'lat' : lat,
+                        'lon' : lon,
+                        'exclude' : 'minutely,hourly',
+                        'appid' : self.weather_api_key,
+                        'units' : 'imperial',
+                        'lang' : 'en'
+                        }
+
+                    weatherPages = []
+
+                    async with aiohttp.request("GET", url, params=queryparams) as response:
+                        if (response.status == 200):
+                            res_json = await response.json()
+                            weatherPages = self.getPageData(lat,lon,res_json,city)
+                        
+                        else:
+                            embed = discord.Embed(title='OpenWeatherMap Weather', color=Colors.Error)
+                            ErrMsg = f'Error Code: {response.status}'
+                            embed.add_field(name='Error with weather API', value=ErrMsg, inline=False)
+                            await ctx.send(embed=embed)
+                            return
+
+                        await self.bot.messenger.publish(Events.on_set_pageable,
+                            embed_name = 'OpenWeatherMap Weather',
+                            field_title = 'Current Conditions and 7-Day Forecast',
+                            pages = weatherPages,
+                            author = ctx.author,
+                            channel = ctx.channel)
+                
+                else:
+                    embed = discord.Embed(title='OpenWeatherMap Weather', color=Colors.Error)
+                    ErrMsg = f'Error Code: {response.status}'
+                    embed.add_field(name='Error with geocode API', value=ErrMsg, inline=False)
+                    await ctx.send(embed=embed)
+                    return
+                """ 
```

Comment:
this was the api latency discussion right

Reply:
Yeah, I forgot to put in the status message before I sent the pull request. It'll be in the next updated commit.